### PR TITLE
Feature: Add toYaml function to templater.

### PR DESCRIPTION
**Pull request by mrmm**
This function will allow to convert interface to Yaml, thus you can use indent function to include keys directly from your ResourcesSets values.

Example Usage
Lets say that you have this ResourcesSets definition:
```yaml
- name: test
  values:
    envVars:
    - name: VALUE_A
      valueFrom:
        configMapKeyRef:
          name: configmap-1
          key: value-a
    - name: RUN_TEST
      value: "true"
```
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod-example
spec:
  containers:
  - name: ubuntu
    image: ubuntu:trusty
    env:
    {{- toYaml .envVars | indent 2 }}
    command: ["echo"]
    args: ["Hello World"]
```
@tazjin could you take a look at this and include it in the next release.

Cheers.

### CODE REVIEW

**Comment by mrmm** [comment]
Yes, I am totally that with this example you can achieve the same using `toJson` function.
For better debugging and visibility it would be helpful to have all your manifest in YAML.

Cheers.

**Comment by tazjin** [comment]
Sorry for the delay, taking another look now!

@mrmm You will need to rebase this on `master`.

File: templater/templater.go [hunk 1]
```diff
@@ -185,6 +186,15 @@ func templateFuncs(c *context.Context, rs *context.ResourceSet) template.FuncMap
 		b, _ := json.Marshal(data)
 		return string(b)
 	}
+
+	m["toYaml"] = func(v interface{}) string {
+		data, err := yaml.Marshal(v)
+		if err != nil {
+			// Swallow errors inside of a template.
+			return ""
```
Comment (tazjin) [inline templater/templater.go]: Template functions are allowed to return errors (see some of the functions below for example) and should do so - we don't want a situation in which templating silently fails.

File: templater/templater.go [hunk 2]
```diff
@@ -185,6 +186,15 @@ func templateFuncs(c *context.Context, rs *context.ResourceSet) template.FuncMap
 		b, _ := json.Marshal(data)
 		return string(b)
 	}
+
+	m["toYaml"] = func(v interface{}) string {
```
Comment (tazjin) [inline templater/templater.go]: I'm still somewhat concerned with the "footgun potential" of having a YAML-serialising function - most Kontemplate outputs are not intended for human consumptions, and for machines it is much more useful to have inline JSON.

I'll need to discuss this with a few people before making a decision.

File: templater/templater.go (hunk 1 above)
Comment (mrmm) [reply templater/templater.go]: Right away ;)

**Comment by tazjin** [comment]
Closing as this has become stale.

**Pull request closed without merging**